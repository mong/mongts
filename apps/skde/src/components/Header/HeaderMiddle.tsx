import React from "react";
import { Toolbar, Typography, styled, Container } from "@mui/material";
import Grid from "@mui/material/Grid2";
import { Breakpoint } from "@mui/material";
import { useRouter } from "next/router";

const StyledToolbar = styled(Toolbar)(({ theme }) => ({
  paddingTop: theme.spacing(12),
  paddingBottom: theme.spacing(8),
}));

export type HeaderData = {
  title: string;
  subtitle: string;
};

type HeaderMiddleProps = {
  headerData: HeaderData;
  children?: React.ReactNode;
  bgcolor?: string;
  maxWidth?: false | Breakpoint;
};

export const HeaderMiddle = (props: HeaderMiddleProps) => {
  const router = useRouter();
  const mainUrl = "https://apps.skde.no" + router.asPath;

  return (
    <StyledToolbar
      sx={{
        bgcolor: process.env.NEXT_PUBLIC_VERIFY
          ? "error.main"
          : props.bgcolor || "primary.light",
      }}
      className="header-middle"
    >
      <Container maxWidth={props.maxWidth} disableGutters={true}>
        <Grid container spacing={2} rowSpacing={6}>
          <Grid size={{ xs: 12 }}>
            {process.env.NEXT_PUBLIC_VERIFY && (
              <>
                <Typography variant="h6">
                  OBS! Dette er en verifiseringsside for testdata. Hvis du
                  ønsker å se offisielle tall må du gå til{" "}
                  <a href={mainUrl} target="_self">
                    www.skde.no.
                  </a>
                </Typography>
                <br />
              </>
            )}
            <Typography variant="h1">{props.headerData.title}</Typography>
          </Grid>
          <Grid size={{ xs: 12 }}>
            <Typography variant="h6">{props.headerData.subtitle}</Typography>
          </Grid>
          {props.children !== undefined && props.children}
        </Grid>
      </Container>
    </StyledToolbar>
  );
};
