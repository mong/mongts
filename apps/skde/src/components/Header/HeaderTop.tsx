import { NavigateNextRounded, ArrowBack } from "@mui/icons-material";
import {
  Breadcrumbs,
  Breakpoint,
  Link,
  Toolbar,
  Typography,
  styled,
  Container,
} from "@mui/material";
import Grid from "@mui/material/Grid2";
import { useTheme } from "@mui/material/styles";
import { useMediaQuery } from "@mui/material";

const StyledToolbar = styled(Toolbar)(({ theme }) => ({
  backgroundColor: theme.palette.background.default,
  paddingTop: theme.spacing(5),
  paddingBottom: theme.spacing(5),
}));

const LogoImage = styled("img")(({ theme }) => ({
  [theme.breakpoints.down("lg")]: {
    height: 40,
    width: 100,
  },
  [theme.breakpoints.up("lg")]: {
    height: 50,
    width: 125,
  },
}));

const StyledBreadcrumbSeparator = styled(NavigateNextRounded)(({ theme }) => ({
  color: theme.palette.primary.light,
}));

type BreadCrumbStop = {
  link: string;
  text: string;
};

export type BreadCrumbPath = {
  path: BreadCrumbStop[];
};

const SkdeBreadcrumbs = (props: BreadCrumbPath) => {
  const { path } = props;

  const theme = useTheme();
  const onMobile = useMediaQuery(theme.breakpoints.down("md"));

  if (onMobile && path.length > 1) {
    const secondToLastElement = path.at(-2);
    return (
      <Breadcrumbs
        separator={<ArrowBack fontSize="medium" />}
        aria-label="breadcrumb"
      >
        <div></div>
        <Link
          underline="hover"
          key="mobile_breadcrumb"
          href={secondToLastElement.link}
          variant="h6"
        >
          {secondToLastElement.text}
        </Link>
      </Breadcrumbs>
    );
  }
  return (
    <Breadcrumbs
      separator={<StyledBreadcrumbSeparator fontSize="large" />}
      aria-label="breadcrumb"
    >
      {path.slice(0, -1).map((row, index) => (
        <Link underline="hover" key={index} color="inherit" href={row.link}>
          {row.text}
        </Link>
      ))}
      <Typography color="text.primary">{path.at(-1).text}</Typography>
    </Breadcrumbs>
  );
};

type HeaderTopProps = {
  breadcrumbs: BreadCrumbPath;
  maxWidth?: false | Breakpoint;
};

export const HeaderTop = (props: HeaderTopProps) => {
  const { breadcrumbs, maxWidth } = props;

  return (
    <StyledToolbar className="header-top">
      <Container maxWidth={maxWidth ? maxWidth : false} disableGutters={true}>
        <Grid container spacing={2}>
          <Grid size={{ xs: 12 }}>
            <Link href={"https://www.skde.no/"}>
              <LogoImage src="/img/logos/logo-skde.svg" alt="SKDE logo" />
            </Link>
          </Grid>
          <Grid size={{ xs: 12 }}>
            <SkdeBreadcrumbs path={breadcrumbs.path} />
          </Grid>
        </Grid>
      </Container>
    </StyledToolbar>
  );
};
