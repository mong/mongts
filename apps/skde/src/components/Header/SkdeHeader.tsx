import { NavigateNextRounded } from "@mui/icons-material";
import { Breadcrumbs, Link, Toolbar, Typography, styled } from "@mui/material";
import Grid from "@mui/material/Unstable_Grid2";

const StyledToolbar = styled(Toolbar)(({ theme }) => ({
  backgroundColor: theme.palette.background.default,
  paddingTop: theme.spacing(5),
  paddingBottom: theme.spacing(5),
}));

const LogoImage = styled("img")(({ theme }) => ({
  [theme.breakpoints.down("lg")]: {
    height: 40,
    width: 100,
  },
  [theme.breakpoints.up("lg")]: {
    height: 50,
    width: 125,
  },
}));

const StyledBreadcrumbSeparator = styled(NavigateNextRounded)(({ theme }) => ({
  color: theme.palette.primary.light,
}));

const breadcrumbsMapping = {
  forside: {
    link: "https://www.skde.no",
    text: "Forside",
  },
  resultater: {
    link: "https://www.skde.no/resultater",
    text: "Tall om helsetjenesten",
  },
  pasientstrømmer: {
    link: "/pasientstrommer/",
    text: "Pasientstrømmer",
  },
  behandlingskvalitet: {
    link: "/behandlingskvalitet/",
    text: "Behandlingskvalitet",
  },
  sykehusprofil: {
    link: "/sykehusprofil/",
    text: "Sykehusprofil",
  },
};

const SkdeBreadcrumbs = ({ path }: { path: string[] }) => {
  return (
    <Breadcrumbs
      separator={<StyledBreadcrumbSeparator fontSize="large" />}
      aria-label="breadcrumb"
    >
      {["forside", ...path].slice(0, -1).map((id, index) => (
        <Link
          underline="hover"
          key={index}
          color="inherit"
          href={breadcrumbsMapping[id].link}
        >
          {breadcrumbsMapping[id].text}
        </Link>
      ))}
      <Typography color="text.primary">
        {breadcrumbsMapping[path.at(-1)].text}
      </Typography>
    </Breadcrumbs>
  );
};

export const SkdeHeader = ({ path }: { path: string[] }) => {
  return (
    <StyledToolbar className="header-top">
      <Grid container spacing={2}>
        <Grid xs={12}>
          <Link href={"/"}>
            <LogoImage src="/img/logos/skde-blue.png" alt="SKDE logo" />
          </Link>
        </Grid>
        <Grid xs={12}>
          <SkdeBreadcrumbs path={path} />
        </Grid>
      </Grid>
    </StyledToolbar>
  );
};

export default SkdeHeader;
