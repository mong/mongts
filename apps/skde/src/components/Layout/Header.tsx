import React from "react";
import Link from "next/link";
import Image from "next/image";

import { imgLoader } from "../../helpers/functions";
import classNames from "./Header.module.css";

import { MenuButton } from "../Btns/MenuButton";
import { Menu } from "../Menu";
import { PopUp } from "../popup";

type HeaderProps = {
  origin: string;
  lang: "no" | "en";
};

export const Header: React.FC<HeaderProps> = ({ origin, lang }) => {
  return (
    <header className={`${classNames.headerContainer}`}>
      <div className={classNames.headerLogo}>
        <Link
          href={`${origin}${lang === "en" ? "/helseatlas/en" : "/helseatlas"}`}
        >
          <a rel="home">
            <Image
              loader={imgLoader}
              src={`/helseatlas/img/logos/helseatlas.svg`}
              alt="Hjem"
              width={186}
              height={47}
            />
          </a>
        </Link>
      </div>
      <nav className={classNames.headerNavContainer}>
        <div className={classNames.navButtons}>
          <div className={classNames.lang}>
            <div
              className={`${classNames.no} ${
                lang === "no" ? classNames.active : undefined
              }`}
              data-testid="buttonNo"
            >
              <Link href={`${origin}/helseatlas`}>
                <a>NO</a>
              </Link>
            </div>
            <div
              className={`${classNames.eng} ${
                lang === "en" ? classNames.active : undefined
              }`}
              data-testid="buttonEng"
            >
              <Link href={`${origin}/helseatlas/en`}>
                <a>ENG</a>
              </Link>
            </div>
          </div>
          <div className={classNames.menuButton}>
            <PopUp btnComponent={() => <MenuButton lang={lang} />}>
              <Menu lang={lang} />
            </PopUp>
          </div>
        </div>
      </nav>
    </header>
  );
};
