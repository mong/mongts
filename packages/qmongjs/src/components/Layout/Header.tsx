import React from "react";
import style from "./Header.module.css";
import skdeLogo from "../../img/logos/SKDE_sort.png";
import Link from "next/link";
import Image from "next/image";
import { imgLoader } from "../../helpers/functions";

export const Header: React.FC = () => {
  return (
    <div style={{}}>
      <header className={style.header}>
        <div className={style.logo}>
          <Link href="https://www.skde.no/">
            <>
              <Image
                loader={imgLoader}
                className={style.logo_img}
                src={skdeLogo}
                alt="SKDE logo"
                height={39}
                width={96}
              />
            </>
          </Link>
        </div>
        <div>
          <Link href={`/kontakt`}>Kontakt</Link>
        </div>
      </header>
    </div>
  );
};
