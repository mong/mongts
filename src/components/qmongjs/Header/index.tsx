import React from "react";
import style from "./index.module.css";
const skdeLogo = "/img/logos/SKDE_sort.png";

type HeaderProps = {};

const Header: React.FC<HeaderProps> = () => {
  const [origin, setOrigin] = React.useState("");
  React.useEffect(() => {
    setOrigin(window.location.origin);
  }, [setOrigin]);
  return (
    <div style={{}}>
      <header className={style.header}>
        <div className={style.logo}>
          <a href={`${origin}`}>
            <img className={style.logo_img} src={skdeLogo} alt="SKDE logo" />
          </a>
        </div>
        <div>
          <a href={`${origin}/kontakt`}>Kontakt</a>
        </div>
      </header>
    </div>
  );
};

export default Header;
