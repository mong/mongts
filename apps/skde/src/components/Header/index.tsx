import React from "react";
import { HeaderTop, BreadCrumbPath } from "./HeaderTop";
import { HeaderMiddle, HeaderData } from "./HeaderMiddle";
import { Breakpoint } from "@mui/material";

type HeaderProps = {
  headerData: HeaderData;
  breadcrumbs: BreadCrumbPath;
  bgcolor?: string;
  maxWidth?: false | Breakpoint;
};

export const Header = (props: HeaderProps) => {
  return (
    <>
      <HeaderTop breadcrumbs={props.breadcrumbs} maxWidth={props.maxWidth} />
      <HeaderMiddle
        bgcolor={props.bgcolor}
        headerData={props.headerData}
        maxWidth={props.maxWidth ? props.maxWidth : false}
      />
    </>
  );
};

export { HeaderTop } from "./HeaderTop";
export type { BreadCrumbPath } from "./HeaderTop";
export type { HeaderData } from "./HeaderMiddle";
