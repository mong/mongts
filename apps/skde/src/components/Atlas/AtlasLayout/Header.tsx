import React from "react";
import Link from "next/link";
import Image from "next/image";

import { imgLoader } from "qmongjs";
import classNames from "./Header.module.css";
import { Box } from "@mui/material";

export const LangButton = ({ lang }: { lang: "no" | "en" }) => {
  return (
    <div className={classNames.lang}>
      <div
        className={`${classNames.no} ${
          lang === "no" ? classNames.active : undefined
        }`}
        data-testid="buttonNo"
      >
        <Link href={`/helseatlas`}>NO</Link>
      </div>
      <div
        className={`${classNames.eng} ${
          lang === "en" ? classNames.active : undefined
        }`}
        data-testid="buttonEng"
      >
        <Link href={`/helseatlas/en`}>ENG</Link>
      </div>
    </div>
  );
};

export const Header = ({ lang }: { lang: "no" | "en" }) => {
  return (
    <header className={`${classNames.headerContainer}`}>
      <Box display="flex" justifyContent="space-between" alignItems="center">
        <div className={classNames.headerLogo}>
          <Link
            href={`${lang === "en" ? "/helseatlas/en" : "/helseatlas"}`}
            rel="home"
          >
            <Image
              loader={imgLoader}
              src={`/helseatlas/img/logos/helseatlas.svg`}
              alt="Hjem"
              width={186}
              height={47}
            />
          </Link>
        </div>
        <LangButton lang={lang} />
      </Box>
    </header>
  );
};
