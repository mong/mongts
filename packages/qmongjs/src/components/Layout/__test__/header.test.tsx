import { render } from "@testing-library/react";
import { it, expect } from "vitest";

import { Header } from "../Header";

it("Header renders", async () => {
  const { container } = render(<Header />);

  expect(container).toMatchSnapshot();
});
