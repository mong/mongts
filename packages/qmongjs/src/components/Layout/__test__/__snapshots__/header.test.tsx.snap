// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main page header renders 1`] = `
<div>
  <div>
    <header
      class="header"
    >
      <div
        class="logo"
      >
        <a
          href="/"
        >
          <img
            alt="SKDE logo"
            class="logo_img"
            data-nimg="1"
            decoding="async"
            height="39"
            loading="lazy"
            src="SKDE_sort.png?w=256&q=75"
            srcset="SKDE_sort.png?w=96&q=75 1x, SKDE_sort.png?w=256&q=75 2x"
            style="color: transparent;"
            width="96"
          />
        </a>
      </div>
      <div>
        <a
          href="/kontakt"
        >
          Kontakt
        </a>
      </div>
    </header>
  </div>
</div>
`;
